src/main/java/com/svydovets/bibirnate/cache/Cache.java

```diff
@@ -0,0 +1,164 @@
+            throw new IllegalArgumentException(
+                    String.format("Inappropriate cacheSize. cacheSize cannot be less then %s or more than %s",
+                            MAX_CACHE_SIZE, MAX_CACHE_SIZE));
+        }
+
+        CACHE_SIZE = cacheSize;
+        lastClean = LocalDateTime.now();
+
+        cacheMap = new ConcurrentHashMap<>(CACHE_SIZE);
+        initializeKeyExtractorCommandMap();
+        initializeInvalidationCommandMap();
+    }
+
+    public void put(Key<?> key, Object value) {
+        Objects.requireNonNull(key, String.format(PARAMETER_CANNOT_BE_NULL, key));
+        Objects.requireNonNull(value, String.format(PARAMETER_CANNOT_BE_NULL, value));
+
+        cacheMap.put(key, value);
+
+        autoClean();
+    }
+
+    public Optional<Object> get(AbstractKeyParam<?> abstractKeyParam, Class<? extends KeyExtractorCommand> commandType) {
+        Objects.requireNonNull(abstractKeyParam, String.format(PARAMETER_CANNOT_BE_NULL, abstractKeyParam));
+        Objects.requireNonNull(commandType, String.format(PARAMETER_CANNOT_BE_NULL, commandType));
+
+        Optional<Key<?>> key = getKey(abstractKeyParam, commandType);
+
+        return key.map(cacheMap::get);
+    }
+
+    public void delete(AbstractKeyParam<?> abstractKeyParam, Class<? extends KeyExtractorCommand> commandType) {
+        Objects.requireNonNull(abstractKeyParam, String.format(PARAMETER_CANNOT_BE_NULL, abstractKeyParam));
+        Objects.requireNonNull(commandType, String.format(PARAMETER_CANNOT_BE_NULL, commandType));
+
+        getKey(abstractKeyParam, commandType).ifPresent(cacheMap::remove);
+    }
+
+    public void invalidateRelated(AbstractKeyParam<?> abstractKeyParam,
+                                  Class<? extends KeyExtractorCommand> extractCommandType,
+                                  Class<? extends InvalidationCommand> invalidationCommandType){
+        Objects.requireNonNull(abstractKeyParam, String.format(PARAMETER_CANNOT_BE_NULL, abstractKeyParam));
+        Objects.requireNonNull(extractCommandType, String.format(PARAMETER_CANNOT_BE_NULL, extractCommandType));
+
+
+        getKey(abstractKeyParam, extractCommandType)
+                .ifPresent(value -> invalidationCommandMap.get(invalidationCommandType)
+                        .executeInvalidate(cacheMap, value));
+    }
+
+    /**
+     * Makes clear for {@link Cache#cacheMap}.
+     */
+    public void clear() {
+        cacheMap.clear();
+    }
+
+    public void addKeyExtractorCommand(KeyExtractorCommand keyExtractorCommand) {
+        Objects.requireNonNull(keyExtractorCommand, String.format(PARAMETER_CANNOT_BE_NULL, keyExtractorCommand));
+
+        keyExtractorCommandMap.put(keyExtractorCommand.getClass(), keyExtractorCommand);
+    }
+
+    private void invalidateRelated(Key<?> key){
+
+    }
+
+    private void autoClean() {
+        if ((Float.valueOf(cacheMap.size()) / Float.valueOf(CACHE_SIZE)) * ONE_HUNDRED > MAX_CACHE_SIZE_PERCENTAGE) {
```

Comment:
Do we really need Float here? I think we can use rough calculations here without float point.
```suggestion
        if ((Float.valueOf(cacheMap.size()) / Float.valueOf(CACHE_SIZE)) * ONE_HUNDRED >= MAX_CACHE_SIZE_PERCENTAGE) {
```

Reply:
I don't know any other ways how to provide what I need. I need to count percentage of how cache map is loaded

Reply:
ArrayList has load factor, you can check how is it implemented.